src/Forms/MultiSelectField.php

```diff
@@ -72,11 +78,28 @@ public function setValue($value, $obj = null)
         if ($obj instanceof DataObject) {
             $this->loadFrom($obj);
         } else {
-            parent::setValue($value);
+            parent::setValue($value, $obj);
         }
         return $this;
     }
 
+    public function setSubmittedValue($value, $data = null)
+    {
+        // When no value is selected and a regular edit form is submitted,
+        // usually an empty string will be POSTed e.g. MyField[]: ""
+        if ($value === [""]) {
```

Comment:
```suggestion
        if ($value === ['']) {
```
Subjective stylistic change in this case, but we do tend to use the safer `''` whenever we're not injecting variables into strings.

Reply:
Updated